### DISCUSSION TITLE
Copilot throws LSP error in Neovim

### DISCUSSION THREAD

**Discussion by mskelton** [community/community#53484] [Copilot Conversations]
### Select Topic Area

Bug

### Body

The Neovim Copilot plugin throws an LSP error very consistently using the following steps.

1. In a project create an empty file (e.g. `test.ts` or `test.go`)
2. Open Neovim (`nvim`)
3. Open the file (`:e test.ts` or `:e test.go`)
4. Insert a new line using the `o` key
5. Notice the following LSP error using `:LspLog`

```
[ERROR][2023-04-22 07:46:31] ...lsp/handlers.lua:535    "Notification handler 'textDocument/didChange' failed with message: Cannot read properties of undefined (reading 'offsetAt')"
```

A few things to note about this error:

- This doesn't happen if you open the file from the command line (e.g. `nvim test.go`)
- The file contents don't matter, the example I gave is just simple, but it also happens if there is content.
- This oddly does not happen when simply entering insert mode with `i`.

I'd love to fix this and submit a PR, but it's in the minified `agent.js` file, so in addition to not being very feasible to debug in that file, submitting a PR is not useful anyway.

**Comment by aeciolevy** [comment]
I am having an issue as well, and cannot fix it.
I would love to know how to debug better these issues.

Every time I open a file
```
LSP[copilot]: Error SERVER_REQUEST_HANDLER_ERROR: "vim/keymap.lua:0: rhs: expected string|function, got nil"
Error detected while processing function <lambda>8[1]..<SNR>44_GetCapabilitiesResult[3]..<SNR>44_LspRequest[1]..LspAttach Autocommands for "*":
Error executing lua callback: vim/keymap.lua:0: rhs: expected string|function, got nil
```
